ddtrace/tracer/tracer_test.go

```diff
@@ -1014,6 +1014,93 @@ func TestTracerReportsHostname(t *testing.T) {
 	})
 }
 
+func TestEnvTags(t *testing.T) {
+	t.Run("service", func(t *testing.T) {
+		os.Setenv("DD_SERVICE", "servenv")
+		defer os.Unsetenv("DD_SERVICE")
+
+		tracer, _, _, stop := startTestTracer(t)
+		defer stop()
+
+		assert := assert.New(t)
+		sp := tracer.StartSpan("http.request").(*span)
+		assert.Equal("servenv", sp.Service)
+	})
+
+	t.Run("version", func(t *testing.T) {
+		os.Setenv("DD_VERSION", "1.2.3")
+		defer os.Unsetenv("DD_VERSION")
+
+		tracer, _, _, stop := startTestTracer(t)
+		defer stop()
+
+		assert := assert.New(t)
+		sp := tracer.StartSpan("http.request").(*span)
+		v := sp.Meta[ext.Version]
+		assert.Equal("1.2.3", v)
+	})
+
+	t.Run("noversion", func(t *testing.T) {
+		os.Setenv("DD_VERSION", "1.2.3")
+		defer os.Unsetenv("DD_VERSION")
+
+		tracer, _, _, stop := startTestTracer(t, WithService("servenv"))
+		defer stop()
+
+		assert := assert.New(t)
+		sp := tracer.StartSpan("http.request", ServiceName("otherservenv")).(*span)
+		_, ok := sp.Meta[ext.Version]
+		assert.False(ok)
+	})
+
+	t.Run("noversion2", func(t *testing.T) {
```

Comment:
I feel like this test and all of the ones below it are not needed. They are basically testing that `DD_SERVICE` properly sets the service (otherwise things wouldn't work). Shouldn't that be a test just for `option.go`?

Reply:
You're right, these probably belong in option.go. I think they are valuable because the test that the version tag is or isn't set in given situations. Maybe there are too many, but there is a combinatorial explosion because of how things can be set. 

If you'd prefer, I can combine them into one test case, but I thought having more, smaller cases was clearer. I can get rid of the `"service"` test, since that is not functionality that's being added/changed in this PR anymore, and it's tested in option_test.go